src/mesh/tdycoremesh.c

```diff
@@ -2268,14 +2268,92 @@ PetscBool AreCellsNeighbors(TDy tdy, PetscInt cell_id_1, PetscInt cell_id_2) {
 }
 
 /* -------------------------------------------------------------------------- */
-PetscErrorCode ArrangeCellsInAnCircularOrder(TDy tdy, PetscInt *cellsAbvBlw, PetscInt ncells, PetscInt *cell_order) {
+PetscErrorCode ArrangeCellsInCircularOrder(TDy tdy, PetscInt *cellsAbvBlw, PetscInt ncells, PetscInt *cell_order) {
 
   PetscFunctionBegin;
+
+  // 1. If all cells in cellsAbvBlw do not have same number of neighbors, then
+  //    - Check to ensure only two cells have miniumum number of neighbors, and
+  //    - Arrange cellsAbvBlw such that cells with minimum number of neighbors are
+  //      at the start and end of the array.
+
+  PetscInt nNeighbors[ncells];
+  for (PetscInt ii = 0; ii < ncells; ii++) {
+    nNeighbors[ii] = 0;
+  }
+
+  // Determine the number of neigbhors
+  for (PetscInt ii = 0; ii < ncells; ii++){
+    for (PetscInt jj = ii+1; jj < ncells; jj++){
+      nNeighbors[ii] += AreCellsNeighbors(tdy, cellsAbvBlw[ii], cellsAbvBlw[jj]);
```

Comment:
If you wanted, you could store the result of `AreCellsNeighbors` here and then add it to both `nNeighbors[ii]` and `nNeighbors[jj]`.